src/Adapter/Form/ChoiceProvider/InstalledPaymentModulesChoiceProvider.php

```diff
@@ -53,4 +76,24 @@ public function getChoices(): array
 
         return self::$paymentModules;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChoicesAttributes()
+    {
+        $attr = [];
+        $orderStates = $this->orderStateDataProvider->getOrderStates($this->contextLangId);
+        foreach (array_keys($this->getChoices()) as $moduleName) {
+            $attr[$moduleName] = array_reduce($orderStates, function ($carry, $item) use ($moduleName) {
```

Comment:
Correct me if I'm wrong, this gives you the state assigned to a module, but it doesn't ensure that it's a "waiting payment" state, because a module can have several states assigned to it

From what I see, this information is in `order_state_lang.template` which needs to be `payment` because it's the email sent to require payment from the customer, but another criteria might be to check for the state flags and search for one that as `paid = 0`